src/mcpdotnet/Configuration/McpServerBuilderExtensions.Tools.cs

```diff
@@ -53,22 +92,7 @@ public static IMcpServerBuilder WithTools(this IMcpServerBuilder builder, params
 
         foreach (var type in toolTypes)
         {
-            var methods = type.GetMethods(BindingFlags.Public | BindingFlags.Instance | BindingFlags.Static);
-            foreach (var method in methods)
-            {
-                var attribute = method.GetCustomAttribute<McpToolAttribute>();
-                if (attribute != null)
-                {
-                    var tool = CreateTool(method, attribute);
-                    tools.Add(tool);
-
-                    callbacks.Add(tool.Name, async (request, cancellationToken) => await CallTool(request, method, cancellationToken).ConfigureAwait(false));
-
-                    // register type because method is not static and so we need an instance
-                    if (!method.IsStatic)
-                        builder.Services.AddScoped(type);
-                }
-            }
+            ProcessToolType(type, builder, JsonSerializerOptionsExtensions.DefaultOptions, tools, callbacks);
```

Comment:
In my PR, I've changed all of this implementation to just use AIFunctionFactory.